client/my-sites/hosting/main.js

```diff
@@ -35,13 +43,35 @@ const DataSection = ( { title, data } ) => {
 	);
 };
 
-const Hosting = ( { translate } ) => {
+const Hosting = ( { translate, siteId } ) => {
+	const [ loadingToken, setLoadingToken ] = useState( false );
+
 	const dummyInfo = {
 		[ translate( 'URL' ) ]: 'sftp1.wordpress.com',
 		[ translate( 'Port' ) ]: 22,
 		[ translate( 'Username' ) ]: 'test',
 		[ translate( 'Password' ) ]: 'test',
 	};
+	function handleClick() {
+		setLoadingToken( true );
+		// We don't want the token to pass through the redux store, so just doing a raw wpcom.req here
+		// TODO: This needs to be a POST, but setting apiNamespace is not currently working for POST
+		wpcom.req
+			.get( `/sites/${ siteId }/hosting/pam/token`, {
+				apiNamespace: 'wpcom/v2',
+			} )
+			.then( response => {
+				if ( response.token ) {
+					window.open( `https://speciallink/pma.php?token=${ response.token }`, '_blank' );
+				}
+			} )
+			.catch( () => {
```

Comment:
Note that chaining instead of providing 2 function arguments to `.then` will also catch accidental errors in the success function.

Reply:
true, been a long time since I used a good old fashioned promise!